client/src/components/BadReview/BadReviewControllers.tsx

```diff
@@ -0,0 +1,110 @@
+import { Select, Row, Button, Col, Modal, message, Spin } from 'antd';
+import { useMemo, useState, useEffect } from 'react';
+import { CheckService } from 'services/check';
+import { CourseTaskDetails } from 'services/course';
+import { BadReviewTable } from './BadReviewTable';
+
+interface IBadReviewControllersProps {
+  courseTasks: CourseTaskDetails[];
+}
+
+export interface IBadReview {
+  checkerScore: number;
+  comment?: string;
+  taskName: string;
+  checkerGithubId: string;
+  studentGithubId: string;
+  studentAvgScore?: number;
+}
+
+export type checkType = 'Bad comment' | 'Did not check' | 'No type';
+
+export function BadReviewControllers({ courseTasks }: IBadReviewControllersProps) {
+  const { Option } = Select;
+  const [taskId, setTaskId] = useState<number>();
+  const [data, setData] = useState<IBadReview[]>();
+  const [isLoading, setIsLoading] = useState<boolean>(true);
+  const [checkType, setCheckType] = useState<checkType>();
+  const [isModalVisible, setIsModalVisible] = useState<boolean>(false);
+  const checkService = useMemo(() => new CheckService(), []);
+
+  const showModal = () => {
+    setIsModalVisible(true);
+  };
+
+  const handleOk = () => {
+    setIsModalVisible(false);
+  };
+
+  const handleCancel = () => {
+    setIsModalVisible(false);
+  };
+
+  const buttonHandler = (type: checkType) => {
+    setCheckType(type);
+    showModal();
+  };
+
+  useEffect(() => setCheckType('No type'), [taskId]);
+
+  useEffect(() => {
+    async function getData(): Promise<void> {
+      if (taskId) {
+        setIsLoading(true);
+        let dataFromService: IBadReview[] = [];
+        try {
+          switch (checkType) {
+            case 'Bad comment':
+              dataFromService = await checkService.getBadComments(taskId, checkType);
+              break;
+            case 'Did not check':
+              dataFromService = await checkService.getMaxScoreCheckers(taskId, checkType);
+              break;
```

Comment:
Better make it as a part of `checkService`

Reply:
Method that should call getBadComments or getMaxScoreCheckers depends on type...

Reply:
+